extendr-api/src/scalar/macros.rs

```diff
@@ -121,6 +121,98 @@ macro_rules! gen_binop {
     };
 }
 
+/// Generates binary operate-assign operators for scalar types.
+macro_rules! gen_binopassign {
+    ($type : tt, $type_prim : tt, $opname : ident, $expr: expr, $docstring: expr) => {
+        impl $opname<$type> for $type {
+            paste::paste! {
+                #[doc = $docstring]
+                fn [< $opname:snake >](&mut self, other: $type) {
+                    match (self.clone().into(), other.into()) {
+                        (Some(lhs), Some(rhs)) => {
```

Comment:
I see this is similar to what we have written for other ops.
However, now I wonder why do we need `.clone()` here.
@andy-thomason, I think you originally wrote this part. Do you have an explanation for this or can we try to remove it and see how it works?

Reply:
If I had to guess, I'd say it's because there's a `TryFrom<>` impl for `Rint`, but not one for `&mut Rint`. Removing the `.clone()` prevents compilation, complaining that `lhs` is an `&mut rint::Rint` and not an `i32`.